util/abs/provision_pe_xl_nodes.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require "./setup/helpers/abs_helper.rb"
+include AbsHelper # rubocop:disable Style/MixinUsage
+
+# TODO: specify as env variable or arg?
+ABS_ID = "slv-473"
+
+# TODO: allow different sizes for each node?
+ABS_SIZE = "c5.2xlarge"
+ABS_VOLUME_SIZE = "80"
+
+ROLES_CORE = %w[master
+                puppet_db
+                compiler_a
+                compiler_b].freeze
+
+ROLES_HA = %w[master_replica
+              puppet_db_replica].freeze
+
+# TODO: specify as env variable or arg?
+ROLES = ROLES_CORE + ROLES_HA
+
+# TODO: move to spec when test cases are implemented
+# for now this allows testing of the create_pe_xl_bolt_files method without provisioning
+TEST_HOSTS = [{ role: "puppet_db", hostname: "ip-10-227-3-22.amz-dev.puppet.net" },
+              { role: "compiler_b", hostname: "ip-10-227-1-195.amz-dev.puppet.net" },
+              { role: "puppet_db_replica", hostname: "ip-10-227-3-158.amz-dev.puppet.net" },
+              { role: "master", hostname: "ip-10-227-3-127.amz-dev.puppet.net" },
+              { role: "compiler_a", hostname: "ip-10-227-3-242.amz-dev.puppet.net" },
+              { role: "master_replica", hostname: "ip-10-227-1-82.amz-dev.puppet.net" }].freeze
+
+# Creates the Bolt inventory file (nodes.yaml) and
+# parameters file (params.json) for the specified hosts
+#
+# Note: designed to use the output of provision_hosts_for_roles
+#
+# @author Bill Claytor
+#
+# @param [Array<Hash>] The hosts
+#
+# @example
+#   hosts = provision_hosts_for_roles(roles)
+#   create_pe_xl_bolt_files(hosts)
+#
+# TODO: move to abs_helper or elsewhere?
+# TODO: spec test(s)
+def create_pe_xl_bolt_files(hosts)
+  template_dir = "./util/abs/templates"
+
+  # HA?
+  master_replica = hosts.detect { |m| m[:role] == "master_replica" }
+  suffix = if master_replica
+             "ha"
+           else
+             "no_ha"
+           end
+
+  # TODO: better alternative to having separate files for ha / no ha?
+  nodes_yaml = File.read("#{template_dir}/nodes_#{suffix}.yaml")
+  params_json = File.read("#{template_dir}/params_#{suffix}.json")
+
```

Comment:
And just because I am curious... why do we need two files with pretty much the same info in them?

Reply:
These two files are specified when running the `pe_xl` plan:
https://github.com/reidmv/reidmv-pe_xl/blob/master/documentation/basic_usage.md#basic-usage-instructions

```
 bolt plan run pe_xl \
   --inventory nodes.yaml \
   --modulepath ~/modules \
   --params @params.json
```

Reply:
sounds good.  I added note to one of the tickets that possibly more info could be in the inventory file, to reduce the need for hostnames to be in the params.  The inventory file got a lot of love after this module was written...